### PR TITLE
[refactor] "Generate Missing Function Definitions" should insert method stubs into the class @implementation when there's no appropriate category implementation

### PR DESCRIPTION
**Pull request by hyp**
rdar://32875896

### CODE REVIEW

File: test/Refactor/ImplementDeclaredMethods/implement-declared-methods.m [hunk 1]
```diff
@@ -34,6 +34,7 @@ - (void)someOtherMethod { }
 // CHECK1: "{{.*}}implement-declared-methods.m" "- (void)method { \n  <#code#>;\n}\n\n+ (void)classMethod { \n  <#code#>;\n}\n\n- (void)implementedMethod { \n  <#code#>;\n}\n\n- (void)method:(int)x with:(int)y { \n  <#code#>;\n}\n\n" [[@LINE-1]]:1 -> [[@LINE-1]]:1
```
Comment (vedantk) [inline test/Refactor/ImplementDeclaredMethods/implement-declared-methods.m:34]: Is the '#ifndef NO_IMPL' needed?

File: lib/Tooling/Refactor/ASTStateSerialization.cpp [hunk 2]
```diff
@@ -40,6 +40,8 @@ class USRToDeclConverter
 } // end anonymous namespace
 
 const Decl *PersistentToASTSpecificStateConverter::lookupDecl(StringRef USR) {
+  if (USR.empty())
+    return nullptr;
```
Comment (vedantk) [inline lib/Tooling/Refactor/ASTStateSerialization.cpp:44]: Are empty strings valid USR strings?

Comment (hyp) [reply lib/Tooling/Refactor/ASTStateSerialization.cpp:44]: Not, they just represent a NULL Decl * pointer.

File: test/Refactor/ImplementDeclaredMethods/implement-declared-methods.m (hunk 1 above)
Comment (hyp) [reply test/Refactor/ImplementDeclaredMethods/implement-declared-methods.m:34]: For some of the tests, yes. ( // RUN: clang-refactor-test perform -action implement-declared-methods -selected=all-methods -continuation-file=%S/Inputs/objcClass.m -query-results=query-mix-impl %s -DNO_IMPL -DMIX_IMPL )

**Review by vedantk — APPROVED** [review]
LGTM

**Pull request merged**